app/services/piece-access-level-service.js

```diff
@@ -62,4 +66,24 @@ export default class PieceAccessLevelService extends Service {
       previousPiece = await previousPiece.previousPiece;
     }
   }
+
+  /*
+   * This method is used to update the access levels of the pieces of a subcase's treatments. When a subcase has
+   * been updated to being confidential, all the pieces connected to it through agenda-item-treatments get the
+   * "Ministerraad" access level.
+   */
+  async updateDecisionsAccessLevelOfSubcase(subcase) {
+    const ministerraad = await this.store.findRecordByUri('access-level', CONSTANTS.ACCESS_LEVELS.MINISTERRAAD);
+    const pieces = await this.store.query('piece', {
+      'filter[agenda-item-treatment][subcase][:id:]': subcase.id,
+      include: 'access-level',
+    });
+
+    await Promise.all(pieces.toArray().map((piece) => {
+      if (piece.accessLevel.get('uri') !== ministerraad.uri) {
```

Comment:
Although this will work (using `get('uri')` will immediately resolve since we included the access-level in the query), we have the habit of still awaiting the promise explicitly to avoid the introduction of bugs in the future when someone changes the `include` statement.
```javascript
const accessLevel = await piece.accessLevel;
if (accessLevel.uri !== ministerraad.uri) {
  ...
}
```